### PR TITLE
Introduce SynType.SignatureParameter.

### PR DESCRIPTION
**Pull request by nojaf**
A less impactful version of #13855.
In this PR I'm only adding the signature parameter representation, not touching anything related to `SynArgInfo`.

### CODE REVIEW

**Comment by nojaf** [comment]
Yes, let's totally take this one, @dsyme ready for review.

File: src/Compiler/Service/ServiceParsedInputOps.fs [hunk 1]
```diff
@@ -1661,7 +1661,8 @@ module ParsedInput =
             | SynType.Array (_, t, _)
             | SynType.HashConstraint (t, _)
             | SynType.MeasurePower (t, _, _)
-            | SynType.Paren (t, _) -> walkType t
+            | SynType.Paren (t, _)
```
Comment (dsyme) [inline src/Compiler/Service/ServiceParsedInputOps.fs:1664]: Please also adjust 

* traverseSynType in ServiceParseTreeWalk
* walkType in `GetEntityKind`  (what is that funciton anyway?  It has so much code walking duplication!)

Maybe also remove the default `| _ ->`  pattern matching in those two cases so these missing cases would have been discovered, and check for other missing cases there

**Review by dsyme — CHANGES_REQUESTED** [review]
Just a couple of changes needed, looks really good otherwise

File: src/Compiler/Service/ServiceParsedInputOps.fs (hunk 1 above)
Comment (nojaf) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1671]: Thanks, I've addressed the missing cases.

**Review by dsyme — APPROVED** [review]

**Pull request merged**